perma_web/perma/celery_tasks.py

```diff
@@ -1224,3 +1228,77 @@ def conditionally_queue_internet_archive_uploads_for_date_range(start_date_strin
 
     else:
         logger.info("Skipped the queuing of file upload tasks: max tasks already in progress.")
+
+
+### WACZ CONVERSION ###
+
+def get_warcs():
+    """
+    Gets the sample WARC ids from the CSV file
+    Retrieves the associated file's contents from storage
+    Downloads them to a temp folder
+    """
+    sample_data_guids = os.path.abspath("perma/wacz_experiment/1000-a-guids.csv")
+    warcs_dir = tempfile.mkdtemp()
+
+    with open(sample_data_guids, mode='r') as file:
+        csv_file = csv.reader(file)
+        for lines in csv_file:
+            file_name = f"{lines[2]}/{lines[1]}"
+            file_path = os.path.join(f"{warcs_dir}", lines[1])
+            contents = default_storage.open(file_name).read()
+            with open(file_path, 'wb') as new_f:
+                new_f.write(contents)
+
+    return warcs_dir
+
+@shared_task
+def convert(input_path, output_folder):
+    """
+    Converts WARC file to WACZ
+    Saves file in a temp folder
+    Saves file in storage
+    """
+    input_file_name = input_path.split('/')[-1]
+    output_path = f"{output_folder}/{input_file_name.split('.')[0]}.wacz"
+    js_wacz_call = subprocess.run(["npx", "js-wacz", "create", "-f", input_path, "-o", output_path],
+                                  capture_output=True, check=True, text=True)
+
+    """
+    Added for A24C-RXK4.warc.gz case where js-wacz logs content-length too small
+    even though the conversion succeeds at the end
+    """
+    if 'WACZ file ready' not in js_wacz_call.stdout:
```

Comment:
It does not at the moment, but `js-wacz` should return an exit code that is not 0 when something bad happens. 
This would simplify error handling here, as `subprocess.run()` would raise an exception. That way, you don't even have to check for `strerr`, or the contents of `stdout`. 

I created a PR for `js-wacz` to help with that: https://github.com/harvard-lil/js-wacz/pull/97

Reply:
That's very cool @matteocargnelutti! I will definitely adopt it here when it is merged.